src/lib/services/project-service.ts

```diff
@@ -492,13 +494,44 @@ export default class ProjectService {
         usersAndGroups: IProjectAccessModel,
         createdBy: string,
     ): Promise<void> {
-        return this.accessService.addAccessToProject(
+        await this.accessService.addAccessToProject(
             usersAndGroups.users,
             usersAndGroups.groups,
             projectId,
             roleId,
             createdBy,
         );
+
+        const role = await this.accessService.getRole(roleId);
+        const { users, groups } = await this.getAccessToProject(projectId);
+
+        const addedUsers = users.filter(({ id }) =>
+            usersAndGroups.users.some(({ id: userId }) => userId === id),
+        );
+
+        const addedGroups = groups.filter(({ id }) =>
+            usersAndGroups.groups.some(({ id: groupId }) => groupId === id),
+        );
```

Comment:
Do we need this? I assume this is to filter out users who already had access? I'm not sure what's the intention here because I believe `users` should be the same as `usersAndGroups` (and therefore `addedUsers` should be the same). But I might be wrong

Reply:
I think so. `usersAndGroups` only include the `id`s of users and groups, so this is a way of fetching some extra information like group names, emails, usernames to add to the payload.

Reply:
Yes, but I don't like that apparently we are loading all users and groups to add that information (unless I'm understanding this wrong).
1. I'm not sure we need those details in the payload (ofc it's nice to have, but if it goes against performance I start to doubt it)
2. Maybe we can just query the user ids we have rather than fetching all and filtering out the ones we don't want

Reply:
@ivarconr WDYT? 
1. Should we only include user and group `id`'s in the payload?
2. Should we add user and group store methods to fetch multiple users and groups based on arrays of `id`'s?

Reply:
we should not include users in a group imho

Reply:
@ivarconr We're no longer including the users that currently belong to the group in the payload:
![image](https://user-images.githubusercontent.com/14320932/236474391-db2cf81a-7dc2-4d9c-a610-cafda90ccc93.png)
What I mean is, on the `users` array, do we need to fetch the users info to add it to the payload, or are `id`s enough?

Reply:
After aligning on standup we're keeping it simple and going for `id`'s only for now: https://github.com/Unleash/unleash/pull/3683/commits/ba836681ecbdaa6aa1e504f47ce693b0f6bdbb34

![image](https://user-images.githubusercontent.com/14320932/237031267-8c8346a7-82dd-4142-ad92-c05b94be4b10.png)